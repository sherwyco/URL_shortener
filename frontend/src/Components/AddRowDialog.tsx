import { FormEvent, useState } from 'react'
import Button from '@mui/material/Button'
import TextField from '@mui/material/TextField'
import Dialog from '@mui/material/Dialog'
import DialogActions from '@mui/material/DialogActions'
import DialogContent from '@mui/material/DialogContent'
import DialogContentText from '@mui/material/DialogContentText'
import DialogTitle from '@mui/material/DialogTitle'

interface AddPropTypes {
    handleAddSubmit: (url: string) => void
}

export default function AddRowDialog(props: AddPropTypes) {
    const { handleAddSubmit } = props

    const [open, setOpen] = useState(false)
    const [isValid, setIsValid] = useState(true)
    const [url, setUrl] = useState('')

    const handleClickOpen = () => {
        setOpen(true)
    }

    const handleClose = () => {
        setOpen(false)
    }

    const isValidUrl = (url: string) => {
        try {
            return Boolean(new URL(url))
        } catch (e) {
            return false
        }
    }

    const handleTextFieldChange = (e: FormEvent<EventTarget>) => {
        const element = e.currentTarget as HTMLInputElement
        const url = element.value
        if (isValidUrl(url)) {
            setUrl(url)
            setIsValid(true)
        } else {
            setIsValid(false)
        }
    }

    const handleSubmit = () => {
        setOpen(false)
        handleAddSubmit(url)
    }

    return (
        <div>
            <Button variant="outlined" onClick={handleClickOpen}>
                Add Row
            </Button>
            <Dialog open={open} onClose={handleClose} fullWidth maxWidth="sm">
                <DialogTitle>Shorten a URL</DialogTitle>
                <DialogContent>
                    <DialogContentText>
                        Please enter the URL that you wish to shorten here.
                    </DialogContentText>
                    <TextField
                        autoFocus
                        margin="dense"
                        id="name"
                        placeholder="https://"
                        error={!isValid}
                        type="url"
                        fullWidth
                        variant="standard"
                        onChange={handleTextFieldChange}
                        helperText="Enter a valid URL"
                    />
                </DialogContent>
                <DialogActions>
                    <Button onClick={handleClose}>Cancel</Button>
                    <Button onClick={handleSubmit} disabled={!isValid}>
                        Submit
                    </Button>
                </DialogActions>
            </Dialog>
        </div>
    )
}
