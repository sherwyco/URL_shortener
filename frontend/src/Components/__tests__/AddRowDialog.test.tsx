import renderer from 'react-test-renderer';
import AddRowDialog from '../AddRowDialog';

it('matches component snapshot', () => {
    const component = renderer.create(<AddRowDialog handleAddSubmit={() => { }} />)
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
})