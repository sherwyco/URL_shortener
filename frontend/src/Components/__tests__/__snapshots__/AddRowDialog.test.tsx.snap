// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches component snapshot 1`] = `
<div>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium css-1rwt2y5-MuiButtonBase-root-MuiButton-root"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    Add Row
  </button>
</div>
`;
